packages/cli/src/commands/devHandler.js

```diff
@@ -23,6 +45,97 @@ export const handler = async ({
 }) => {
   const rwjsPaths = getPaths()
 
+  let apiPort = getConfig().api.port
+  let webPort = getConfig().web.port
+
+  // Check api port
+  if (side.includes('api')) {
+    const freePort = await getFreePort(apiPort)
+    if (freePort === -1) {
+      console.error(
+        c.error(
+          `Requested API port of ${apiPort} is already in use and no neighbouring port is available! Cannot start development server.`
+        )
+      )
+      process.exit(1)
+    }
+    if (freePort !== apiPort) {
+      console.log(
+        c.warning(
+          `Requested API port of ${apiPort} is already in use however ${freePort} is available.`
+        )
+      )
+      const useAvailablePort = await prompts({
+        type: 'confirm',
+        name: 'port',
+        message: `Do you wish to use port ${freePort} instead?`,
+        initial: true,
+        active: 'Yes',
+        inactive: 'No',
+      })
+      if (!useAvailablePort.port) {
+        console.log(c.info(`The API port can be updated in 'redwood.toml'.`))
+        process.exit(0)
+      }
+    }
+    apiPort = freePort
+
+    // TODO: Check the apiDebugPort too?
+  }
+
+  // Check web port
+  if (side.includes('web')) {
+    // Check for specific forwarded web port
+
+    const forwardedPortMatches = forward.match(
+      /--port=[0-9][0-9]?[0-9]?[0-9]?[0-9]? ?/
+    )
+    const forwardedPortSet =
+      forwardedPortMatches && forwardedPortMatches.length == 1
+    if (forwardedPortSet) {
+      webPort = forwardedPortMatches[0]
+        .substring(forwardedPortMatches[0].indexOf('=') + 1)
+        .trim()
+    }
+
+    const freePort = await getFreePort(webPort, [apiPort])
+    if (freePort === -1) {
+      console.error(
+        c.error(
+          `Requested web port of ${webPort} is already in use and no neighbouring port is available! Cannot start development server.`
+        )
+      )
+      process.exit(1)
+    }
+    if (freePort !== webPort) {
+      console.log(
+        c.warning(
+          `Requested web port of ${webPort} is already in use however ${freePort} is available.`
+        )
+      )
+      const useAvailablePort = await prompts({
+        type: 'confirm',
+        name: 'port',
+        message: `Do you wish to use port ${freePort} instead?`,
+        initial: true,
+        active: 'Yes',
+        inactive: 'No',
+      })
+      if (!useAvailablePort.port) {
+        console.log(
+          c.info(
+            `The web port can be updated in 'redwood.toml' or can be forwarded via the command line parameter like so 'yarn rw dev --fwd="--port=12345"'.`
+          )
+        )
+        process.exit(0)
```

Comment:
(See another similar suggestion)

```suggestion
        process.exit(1)
```